src/ContainerTerminal.jsx

```diff
@@ -0,0 +1,184 @@
+            this.connectChannel();
+        if (prevProps.width !== this.props.width) {
+            this.resize(this.props.width);
+        }
+    }
+
+    resize(width) {
+        var padding = 11 + 5 + 50;
+        var realWidth = this.state.term._core.renderer.dimensions.actualCellWidth;
+        var cols = Math.floor((width - padding) / realWidth);
+        this.state.term.resize(cols, 24);
+        cockpit.spawn(["sh", "-c", "echo '1 24 " + cols.toString() + "'>" + this.state.control_channel], { superuser: true });
+        this.setState({ cols: cols });
+    }
+
+    connectChannel() {
+        let self = this;
+        if (self.state.channel)
+            return;
+
+        if (self.props.containerStatus !== "running") {
+            let message = _("Container is not running");
+            this.setState({ errorMessage: message });
+            return;
+        }
+
+        varlink.call(utils.PODMAN_ADDRESS, "io.podman.GetAttachSockets", { name: this.state.container })
+                .then(out => {
+                    let opts = {
+                        payload: "packet",
+                        unix: out.sockets.io_socket,
+                        superuser: "require",
+                        binary: false
+                    };
+
+                    let channel = cockpit.channel(opts);
+                    channel.wait()
+                            .then(() => {
+                                // Show the terminal. Once it was shown, do not show it again but reuse the previous one
+                                if (!this.state.opened) {
+                                    this.state.term.open(this.refs.terminal);
+                                    this.setState({ opened: true });
+
+                                    self.state.term.on('data', function(data) {
+                                        if (self.state.channel)
+                                            self.state.channel.send(data);
+                                    });
+                                }
+
+                                channel.addEventListener("message", this.onChannelMessage);
+                                channel.addEventListener('close', this.onChannelClose);
+
+                                channel.send(String.fromCharCode(12)); // Send SIGWINCH to show prompt on attaching
+                                this.setState({ channel: channel, control_channel: out.sockets.control_socket, errorMessage: "" });
+                                this.resize(this.props.width);
+                            })
+                            .catch(e => {
+                                let message = cockpit.format(_("Could not open channel: $0"), e.problem);
+                                if (e.problem === "not-supported")
+                                    message = _("Upgrade to newer version of the Web Console for an interactive terminal");
```

Comment:
Yesterday we discussed that a bit, and this may lead to confusion. E. g. in RHEL 8.1 or Fedora 29 there isn't really a newer version to upgrade to. How about "This version of the Web Console does not support a terminal."?

Reply:
Right, this should not cause confusion when users try to update cockpit and won't find any newer version. Updated.